spec/content/automate/ManageIQ/Transformation/Ansible.class/__methods__/launchplaybookasaservice_spec.rb

```diff
@@ -0,0 +1,135 @@
+require_domain_file
+
+RSpec::Matchers.define :same_class_and_id do |obj|
+  match { |actual| actual.class == obj.class && actual.id == obj.id }
```

Comment:
@fdupont-redhat I have been looking into this more over the weekend and think we should override the `==` method on the [class MiqAeServiceModelBase](https://github.com/ManageIQ/manageiq-automation_engine/blob/master/lib/miq_automation_engine/engine/miq_ae_method_service/miq_ae_service_model_base.rb#L4) to use the same logic as this matcher then the test would just need to use `eq`.

```ruby
    def ==(other)
      self.id == other.id && self.class == other.class
    end
```

See also [BasicObject#==](https://ruby-doc.org/core-2.5.0/BasicObject.html#method-i-3D-3D)
> Equality — At the Object level, == returns true only if obj and other are the same object. Typically, this method is overridden in descendant classes to provide class-specific meaning.

cc @mkanoor

Reply:
I proposed https://github.com/ManageIQ/manageiq-automation_engine/pull/348